vulnerabilities/pipelines/v2_importers/project_kb_importer.py

```diff
@@ -0,0 +1,194 @@
+        self.vuln_data_branch_vcs = fetch_via_vcs(self.vuln_data_branch)
+
+    def advisories_count(self):
+        base_path = Path(self.vuln_data_branch_vcs.dest_dir) / "statements"
+        csv_path = (
+            Path(self.main_branch_vcs.dest_dir) / "MSR2019/dataset/vulas_db_msr2019_release.csv"
+        )
+
+        count_files = sum(1 for _ in base_path.rglob("*.yaml"))
+        with open(csv_path, newline="", encoding="utf-8") as f:
+            reader = csv.reader(f)
+            next(reader, None)
+            count_rows = sum(1 for _ in reader)
+
+        count = count_files + count_rows
+        self.log(f"Estimated advisories to process: {count}")
+        return count
+
+    def collect_advisories(self) -> Iterable[AdvisoryData]:
+        self.log("Collecting fix commits from YAML statements under /statements....")
+        base_path = Path(self.vuln_data_branch_vcs.dest_dir) / "statements"
+
+        for yaml_file in base_path.rglob("*.yaml"):
+            if yaml_file.name != "statement.yaml":
+                continue
+
+            with open(yaml_file, encoding="utf-8") as f:
+                yaml_data = saneyaml.load(f)
+
+            vulnerability_id = yaml_data.get("vulnerability_id")
+            if not vulnerability_id:
+                continue
+
+            note_texts = []
+            for note_entry in yaml_data.get("notes", []):
+                text_content = note_entry.get("text")
+                if not text_content:
+                    continue
+                note_texts.append(text_content)
+            description = "\n".join(note_texts)
+
+            references = []
+            affected_packages = []
+            patches = []
+            for fix in yaml_data.get("fixes", []):
+                for commit in fix.get("commits", []):
+                    commit_hash = commit.get("id")
+                    if not is_commit(commit_hash):
+                        commit_hash = None
+
+                    vcs_url = commit.get("repository")
+                    append_patch_classifications(
+                        url=vcs_url,
+                        commit_hash=commit_hash,
+                        patch_text=None,
+                        affected_packages=affected_packages,
+                        references=references,
+                        patches=patches,
+                    )
+
+            for artifact in yaml_data.get("artifacts", []):
+                affected = artifact.get("affected")
+                if not affected:
+                    continue
+
+                purl_str = artifact.get("id")
+                purl = PackageURL.from_string(purl_str)
+
+                try:
+                    version_range_class = RANGE_CLASS_BY_SCHEMES.get(purl.type)
+                    version_class = (
+                        version_range_class.version_class if version_range_class else None
+                    )
+                    version_range = version_class(purl.version)
```

Comment:
This is incorrect, it will give `Version` object we need proper `VersionRange` object to populate `AffectedPackageV2`.

Reply:
Yes, you are right not sure why it doesn't throw a error in the logs